parser/helper/helpers.go

```diff
@@ -99,25 +104,32 @@ func (h *Helper) GetActorAddressInfo(add address.Address, key filTypes.TipSetKey
 	return addInfo
 }
 
-func (h *Helper) GetActorNameFromAddress(address address.Address, height int64, key filTypes.TipSetKey) string {
-	// Search for actor in cache
-	actorCode, err := h.actorCache.GetActorCode(address, key)
-	if err != nil {
-		return actors.UnknownStr
-	}
+func (h *Helper) GetActorNameFromAddress(address address.Address, height int64, key filTypes.TipSetKey) (string, error) {
+	onChainOnly := false
+	for {
+		// Search for actor in cache
+		actorCode, err := h.actorCache.GetActorCode(address, key, onChainOnly)
+		if err != nil {
+			return actors.UnknownStr, err
+		}
 
-	c, err := cid.Parse(actorCode)
-	if err != nil {
-		h.logger.Sugar().Errorf("Could not parse params. Cannot cid.parse actor code: %v", err)
-		return actors.UnknownStr
-	}
+		c, err := cid.Parse(actorCode)
+		if err != nil {
+			h.logger.Sugar().Errorf("Could not parse params. Cannot cid.parse actor code: %v", err)
+			return actors.UnknownStr, err
+		}
 
-	actorName, err := h.lib.BuiltinActors.GetActorNameFromCid(c)
-	if err != nil {
-		return actors.UnknownStr
-	}
+		actorName, err := h.lib.BuiltinActors.GetActorNameFromCid(c)
+		if err != nil {
+			return actors.UnknownStr, err
+		}
 
-	return actorName
+		if actorName == manifest.PlaceholderKey && !onChainOnly {
```

Comment:
Is it possible that the condition always evaluates to true and never leave the loop?

Reply:
If this is true the first time (onChainOnly=false), onChainOnly will be true next time, and the loop will be broken!